install/management/appliance.go

```diff
@@ -348,12 +330,19 @@ func (d *Dispatcher) setMacToGuestInfo() error {
 			err = errors.Errorf("Timeout to get VM MAC address")
 			return err
 		}
-		netName := d.nics[key]
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/portgroup", netName), Value: key})
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/mac", netName), Value: value})
-		keys = append(keys, netName)
+		for nicName, netInfo := range conf.Networks {
```

Comment:
The key for the maps is different, yes, but the key for `m` is part of the value for `Networks` (assuming this is the comparison you actually want to make - you say netName in the reply above):
````
if netInfo.PortGroupName == key
```
See https://play.golang.org/p/ks6tpd7vdL